backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/LabelStyle.java

```diff
@@ -165,4 +179,49 @@ public LabelStyle build() {
             return labelDescription;
         }
     }
+
+    /**
+     * Computes the text bounds for a label with the given text.
+     *
+     * @param text
+     *            the text
+     * @return the text bounds
+     */
+    public TextBounds computeBounds(String text) {
+        int fontStyle = Font.PLAIN;
+        if (this.isBold()) {
+            fontStyle = fontStyle | Font.BOLD;
+        }
+        if (this.isItalic()) {
+            fontStyle = fontStyle | Font.ITALIC;
+        }
+        Font font = new Font(DEFAULT_LABEL_FONT_NAME, fontStyle, this.getFontSize());
+        Rectangle2D stringBounds = font.getStringBounds(text, FONT_RENDER_CONTEXT);
+        double width = stringBounds.getWidth();
+        double height = stringBounds.getHeight();
+
+        double iconWidth = 0;
+        double iconHeight = 0;
+        if (!this.getIconURL().isEmpty()) {
+            iconWidth = SPACE_FOR_ICON;
+            if (height < iconWidth / 2) {
+                iconHeight = iconWidth / 2;
+            }
+        }
+
+        // @formatter:off
+            Size size = Size.newSize()
+                    .width(width + iconWidth)
+                    .height(height + iconHeight)
+                    .build();
+
+            // Sprotty needs the inverse of the x and y for the alignment, so it's "0 - x" and "0 - y" on purpose
+            Position alignment = Position.newPosition()
+                    .x(0 - stringBounds.getX() + iconWidth)
+                    .y(0 - stringBounds.getY())
+                    .build();
+            // @formatter:on
+
+        return new TextBounds(size, alignment);
```

Comment:
Never mix data structure and behavior, remove this method from this class. I am quite surprised that the build is not failing on that kind of things

Reply:
Would a LabelStyleHelper with a single static method be mode acceptable here ?

Reply:
fixed with fea91e369c5672d85882a830f4a7a29e509b1352